### DISCUSSION TITLE
Model::find($id) ... phpstan claiming it returns a collection?

### DISCUSSION THREAD

**Discussion by EliW** [larastan/larastan#2059] [Q&A] [answered]
Probably a simple 'duh' ... but the title basically says it all.

When doing a ::find($id) ... against a proper primary-key-id of a Model ... Eloquent is going to only ever return NULL, or the Model.   But phpstan is complaining that it instead is returning Model, or a Collection of Model.

But it won't ever return a collection, since it's a lookup via primary key.   (would be obviously different if you were doing a search/lookup by a different non-unique column)

**Comment by calebdw** [comment, marked as answer]
We have tests for this: https://github.com/larastan/larastan/blob/b0bfa092aa3fac7a5df36ce4fb8cd90866ac8f0e/tests/Type/data/model.php#L68-L74

Could be that you need to narrow the `$id` type

**Reply by EliW** [reply]
Ahhh, that let me right to the issue.   In this case, the actual code was:
`$lease = Lease::find($request->lease_id);`

Changing that to:
`$lease = Lease::find((int)$request->lease_id);`

Immediately solved/cleared all the warnings ...

Though that does bring up an interesting point (not that it matters in this case) ... it implies that your tests are specifically looking  for ::find(int) ... however, a primary_key for a table isn't required to be an integer (even though it usually is).   You could have a varchar() primary key.

And so, ::find(string) (which is what was happening here), should still only ever return 'model|null'.   since `::find($var)` is only ever a primary key lookup, never anything else.

#JustThoughts

**Reply by calebdw** [reply]
> Though that does bring up an interesting point (not that it matters in this case) ... it implies that your tests are specifically looking for ::find(int) ...

This is not true, the extension properly handle the different types. The type of `$request->lease_id` is `mixed` which means we don't know anything about it and therefore return the full type. You need to cast to an int or use `$request->integer('lease_id')`


https://github.com/larastan/larastan/blob/b0bfa092aa3fac7a5df36ce4fb8cd90866ac8f0e/src/ReturnTypes/ModelFindExtension.php#L97-L98

**Reply by EliW** [reply]
Thanks for the clarification!
